src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs

```diff
@@ -39,8 +51,46 @@ public static IApplicationBuilder UseFhirServer(this IApplicationBuilder app, Fu
                 app.UseMiddleware<PathBaseMiddleware>(pathString);
             }
 
+            app.UseHttpsRedirection();
             app.UseStaticFiles();
-            app.UseMvc();
+
+            app.UseRouting();
+            useDevelopmentIdentityProvider?.Invoke(app);
+            useHttpLoggingMiddleware?.Invoke(app);
+
+            app.UseCors(Constants.DefaultCorsPolicy);
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            app.UseEndpoints(
+                endpoints =>
+                {
+                    endpoints.MapControllers();
+                    endpoints.MapHealthChecks(
```

Comment:
Looks like something we can move to shared components later?

Reply:
Good Point! Yes, we can do that in next iteration.